p2p/server.go

```diff
@@ -358,6 +375,27 @@ func (srv *Server) Start() (err error) {
 	if srv.running {
 		return errors.New("server already running")
 	}
+
+	// open mysql db handle
+	if srv.MySQLName != "" {
+		driver := "mysql"
+		db, err := sql.Open(driver, srv.MySQLName)
+		if err != nil {
+			log.Proto("MYSQL", "action", "open handle", "result", "fail", "database", srv.MySQLName, "driver", driver, "err", err)
+			srv.DB = nil
```

Comment:
the Go cannon is to return err here, so that you don't need the `else` conditional, which can lead to deeply nested errors code.

Reply:
done